mephisto/abstractions/providers/prolific/prolific_unit.py

```diff
@@ -0,0 +1,344 @@
+
+    def get_requester(self) -> 'ProlificRequester':
+        """Wrapper around regular Requester as this will be ProlificRequester"""
+        if self.__requester is None:
+            self.__requester = cast('ProlificRequester', super().get_requester())
+        return self.__requester
+
+    def get_status(self) -> str:
+        """Get status for this unit directly from Prolific, fall back on local info"""
+
+        # -------------------------------------------------------------------------
+        # TODO (#1008): Fix this logic. It looked abstract and unclear in Mturk code.
+        if self.db_status == AssignmentState.CREATED:
+            return super().get_status()
+        elif self.db_status in [
+            AssignmentState.ACCEPTED,
+            AssignmentState.EXPIRED,
+            AssignmentState.SOFT_REJECTED,
+        ]:
+            # These statuses don't change with a get_status call
+            return self.db_status
+
+        # -------------------------------------------------------------------------
+        # TODO (#1008): Fix this logic. It looked abstract and unclear in Mturk code.
+        if self.db_status in [AssignmentState.COMPLETED, AssignmentState.REJECTED]:
+            # These statuses only change on agent dependent changes
+            agent = self.get_assigned_agent()
+            found_status = self.db_status
+
+            if agent is not None:
+                agent_status = agent.get_status()
+                if agent_status == AgentState.STATUS_APPROVED:
+                    found_status = AssignmentState.ACCEPTED
+                elif agent_status == AgentState.STATUS_REJECTED:
+                    found_status = AssignmentState.REJECTED
+                elif agent_status == AgentState.STATUS_SOFT_REJECTED:
+                    found_status = AssignmentState.SOFT_REJECTED
+            else:
+                logger.warning(f'Agent for unit {self} is None')
+
+            if found_status != self.db_status:
+                self.set_db_status(found_status)
+
+            return self.db_status
+
+        # Remaining statuses are tracking a live Study
+        prolific_study_id = self.get_prolific_study_id()
+        if prolific_study_id is None:
+            # If the study_id is None and there's an agent still assigned,
+            # then that agent has timed out, and we should expire
+            agent = self.get_assigned_agent()
+            if agent is not None:
+                if agent.get_status() != AgentState.STATUS_EXPIRED:
+                    agent.update_status(AgentState.STATUS_EXPIRED)
+
+            # Can't determine anything else if there is no Study on this unit
+            return self.db_status
+
+        requester: 'ProlificRequester' = self.get_requester()
+        client = self._get_client(requester.requester_name)
+
+        # time.sleep(2)  # Prolific servers may take time to bring their data up-to-date
+        study = prolific_utils.get_study(client, prolific_study_id)
```

Comment:
And you should write `get_submission`, as the statuses of a submission should align here:
<img width="793" alt="Screenshot 2023-07-27 at 11 49 50 AM" src="https://github.com/facebookresearch/Mephisto/assets/1276867/96d33cd5-5aaf-48ae-8604-7d8d688a3dda">